### ISSUE TITLE
Add terminal logging 

### ISSUE THREAD

**Issue by freeseacher** [gnunn1/tilix#1139]
For some reasons i have to log everything inputed with ssh connection. 
Please add logging to file all visible text. 
Files have to be stored in separate directory and named with bookmark name and datetime when session started.

**Comment by gnunn1** [comment]
Tilix has a feature to save the output of the terminal to a file called Save Output. Does that provide what you need?

**Comment by freeseacher** [comment]
i have to do it manualy on each session ?

**Comment by gnunn1** [comment]
That is correct.

I'm not sure from a design perspective how to handle this automatically. You suggested using the Bookmark name for the file but few people use bookmarks. I could dump the logs into a directory with the name being a combination of the terminal uuid and timestamp but that isn't very useful to find a past log.

Do any other terminal emulators do this and if so how do they handle it?

**Comment by freeseacher** [comment]
i am not dividing terminal emulators and connection managers before. 
gcm can do it. putty can do it. 
that is pretty standard functionality for that kind of software.

saving terminal output by hands will lead to lots of manual work. my current archive shows that i'am opening about 40-50 connections a day.

**Comment by aaronSkar** [comment]
I agree, an option for automatic session logs should be implemented.

**Comment by dandv** [comment]
Would also like automatic logging. I'm trying to replace the excellent [ZOC](https://www.emtec.com/zoc/) client for Windows/MacOS.

**Comment by Len4i** [comment]
+1
This is very common feature for terminal emulators and very useful for debug sessions.

**Comment by mahdiadnan** [comment]
+1 for automating logging to file based on session or tab name.

**Comment by sclarkso** [comment]
I would love this feature as well.
